submissions/smillims/oop/app.js

```diff
@@ -0,0 +1,121 @@
+const block = {
+	width: 101,
+	height: 85
+};
+const positionEnemyY = [60, 140, 220];
+const positionPlayer = {
+	playerX: 200,
+	playerY: 400,
+};
+
+const widthEnemies = [80];
+
+// Enemies our player must avoid
+const Enemy = function(x, y) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+
+	 this.x = x;
+	 this.y = y;
+	 this.speed = (block.width * 1.5) + (Math.random() * block.width); //speed between 100 and 250;
+
+    // The image/sprite for our enemies, this uses
+	 // a helper we've provided to easily load images
+	 
+	 this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+	 // all computers.
+	 this.x += this.speed * dt;
+	 if (this.x > ctx.canvas.width + block.width){
+		 this.x = -block.width * 2;
+	 }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.checkCollisions = function() {
+	if(Math.abs(this.x - player.x) < widthEnemies &&
+		Math.abs(this.y - player.y) < widthEnemies ||
+		player.y < 0){
+			setTimeout(() => player.startOver(), 120);
+	};
```

Comment:
Classes should not refer to any global variables directly, like global variable `player`, which is an instance of `Player` class (referring to global constants and globals provided by the gaming platform like `Resources` is OK).

https://github.com/kottans/frontend-2021-homeworks/pull/279#discussion_r546909025

Hint: Check the way `Player` gets to know about `positionPlayer.playerX`

Reply:
I tried to understand how to correct it's problem and solve of this problem was refer to `player` and setup `positionPlayer.playerX`, `positionPlayer.playerX`; if it's not solve for my problem so I don't understand how to fix it if I not need to refer to `player`

Reply:
@smillims let's make a step at a time.
How does Enemy class learn about its `y` position that is defined somewhere outside the class?

Reply:
@OleksiyRudenko it's defined in `positionEnemyY` which defined in `allEnemies` by creating `new Enemy`

Reply:
Not exactly. The more accurate description of the situation would be:
Each Enemy's y position is stored in a global constant `positionEnemyY`, which is passed to an `Enemy` constructor as a parameter by means of `new Enemy`, and in the `Enemy` class constructor the argument for y position gets assigned to an `Enemy`'s own `y` property (which is then referred to as `this.y` in all `Enemy`'s methods).

Do you see a way to handle global `player` properly?

Reply:
by using `this.player.x/y`?

Reply:
Just pass global `player` to each enemy and store it in `this.player`